racket: update to 8.9

OriginCode opened the pull request:
<!-- For description on topic creation and maintenance, please refer to [this Wiki article](https://wiki.aosc.io/developer/packaging/topic-based-maintenance-guideline/). -->

Topic Description
-----------------

<!-- Please input topic description here. -->
Update Racket to 8.9

Package(s) Affected
-------------------

<!-- Please list all package(s) affected by this topic here. -->
`racket` 8.9

Security Update?
----------------

<!-- If this topic contains security update(s), please uncomment "Yes,"
     mark with the `security` and `priority` labels, and make sure to mark your commits to relevant issue numbers.

     Please see GitHub's documentation on "Linking a pull request to an issue":

     https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue -->

<!-- Yes - Issue Number: ISSUENUMBER -->
No

<!-- Please uncomment the "Build Order" section if your topic affects more than one package. -->

Test Build(s) Done
------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->    
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

<!-- Maintainers should review file changes and, if the build script(s) affected complies with the
     [Package Styling Manual](https://wiki.aosc.io/developer/packaging/package-styling-manual/), please
     add `lgtm` label to this issue. -->

<!-- Maintainers and users may now test the packages in this topic and, once user/maintainer feedback indicates that the update(s) work as expected and find its quality satisfactory,
     another maintainer may now review this pull request and mark it as Approved. After which, the maintainer will build affected package(s) and upload them to the `stable` repository. -->

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

<!-- TODO: CI to auto-fill architectural progress. -->

OriginCode committed:
racket: update to 8.9

Signed-off-by: Kaiyang Wu <origincode@aosc.io>

OriginCode committed:
racket: remove redundant dependencies and options

Signed-off-by: Kaiyang Wu <origincode@aosc.io>

OriginCode commented:
LGTM on amd64:
![圖片](https://github.com/AOSC-Dev/aosc-os-abbs/assets/17294071/88135c20-5194-441b-8e6d-163e4e843755)

OriginCode committed:
racket: fix build on loongson3

Signed-off-by: Kaiyang Wu <origincode@aosc.io>

MingcongBai commented on lang-lisp/racket/autobuild/beyond:
```diff
@@ -1,6 +1,7 @@
+abinfo "Installing DrRacket desktop entry and icons ..."
+install -Dvm644 "$SRCDIR"/../share/pkgs/drracket/drracket/drracket.desktop "$PKGDIR"/usr/share/applications/drracket.desktop
+install -dv "$PKGDIR"/usr/share/icons/hicolor/{16x16,32x32,48x48,256x256}/apps
+ln -sv /usr/share/racket/pkgs/icons/plt-16x16.png "$PKGDIR"/usr/share/icons/hicolor/16x16/apps/drracket.png
```
Use relative path here.

MingcongBai committed:
racket: improve beyond

- Install icon symlinks with relative paths to prevent rootfs pollution.
- Use a for loop to simplify repetitive operations.

MingcongBai committed:
racket: set Icon= in .desktop entry
